Move John’s apply form changes from commissions repo to digital repo

jessicamarcus opened the pull request:
_I'm just including the messages from the squashed commits verbatim; I tend to be terse, but am happy to start making my commit messages more conversational if you'd prefer!_

removed node engine declaration from package.json

updated to match changes in deprecated “commissions” repo

added bottom margin, "subheader" variant to component

minor bugfixes, code cleanup

updating snapshots

jessicamarcus committed:
fixes #31

moved commissions edits to digital; cleanup

removed node engine declaration from package.json

updated to match changes in deprecated “commissions” repo

added bottom margin, "subheader" variant to component

minor bugfixes, code cleanup

updating snapshots

jessicamarcus committed:
moved commissions edits to digital; cleanup

removed node engine declaration from package.json

updated to match changes in deprecated “commissions” repo

added bottom margin, "subheader" variant to component

minor bugfixes, code cleanup

updating snapshots

jessicamarcus commented:
Hrm. I thought this PR would be autolinked to the item in Projects by using the item's number, but that doesn't seem to be the case... Can you walk me through how to do this? thx!

fionawhim commented on modules-js/react-fleet/src/SectionHeader.stories.tsx:
```diff
@@ -5,4 +5,5 @@ import SectionHeader from './SectionHeader';
   .add('default', () => <SectionHeader title="Boards and Commissions" />)
-  .add('yellow', () => <SectionHeader title="Boards and Commissions" yellow />);
+  .add('yellow', () => <SectionHeader title="Boards and Commissions" yellow />)
+  .add('subheader', () => <SectionHeader title="Boards and Commissions" subheader />);
```
Nice.

fionawhim commented on services-js/commissions-app/src/pages/commissions/apply.tsx:
```diff
@@ -83,7 +89,7 @@ export default class ApplyPage extends React.Component<Props> {
               email: '',
               confirmEmail: '',
-              commissionIds: [] as string[],
+              commissionIds: [commissionID] as string[],
```
I think this should change to `commissionID ? [commissionID] : []`, and then the `as string[]` can be taken out.

fionawhim commented on services-js/commissions-app/src/pages/commissions/apply.tsx:
```diff
@@ -153,26 +159,31 @@ export default class ApplyPage extends React.Component<Props> {
+
+                  <div className="g--6 m-b300"
+                       style={{
+                         display: 'flex',
```
It would be good to switch this to Emotion so that these get vendor-prefixed as necessary.

jessicamarcus replied:
oops, I am used to autoprefixer in my builds and spaced out on this - thx!

fionawhim commented on services-js/commissions-app/src/pages/commissions/apply.tsx:
```diff
@@ -181,20 +192,23 @@ export default class ApplyPage extends React.Component<Props> {
                       error={touched.firstName && errors.firstName}
                       onBlur={handleBlur}
+                      required
+                      style={{ flexGrow: 1 }}
```
Emotion here as well

fionawhim commented on services-js/commissions-app/src/pages/commissions/apply.tsx:
```diff
@@ -251,8 +272,10 @@ export default class ApplyPage extends React.Component<Props> {
+                      required
                     />
                   </div>
+
```
Good use of whitespace throughout. I 💜blank lines.

fionawhim reviewed (approved):
LGTM!

jessicamarcus committed:
Merge branch 'develop' into code-update-cleanup

jessicamarcus committed:
fixes #31

moved commissions edits to digital; cleanup

removed node engine declaration from package.json

updated to match changes in deprecated “commissions” repo

added bottom margin, "subheader" variant to component

minor bugfixes, code cleanup

updating snapshots
